MusicTaste/MusicTaste/MatchingAlgorithm.m

```diff
@@ -58,35 +54,36 @@ + (double) compareSpotifyData:(PFObject *)potentialMatch withData:(NSManagedObje
                 NSInteger albumCount = [matchDataAlbums count];
                 NSInteger trackCount = [matchDataTracks count];
                 NSInteger artistCount = [matchDataArtists count];
+                //Makes copies of array to avoid deletion of data
+                NSMutableArray *matchDataGenresCopy = [[NSMutableArray alloc] initWithArray:matchDataGenres];
+                NSMutableArray *matchDataTracksCopy = [[NSMutableArray alloc] initWithArray:matchDataTracks];
+                NSMutableArray *matchDataAlbumsCopy = [[NSMutableArray alloc] initWithArray:matchDataAlbums];
+                NSMutableArray *matchDataArtistsCopy = [[NSMutableArray alloc] initWithArray:matchDataArtists];
                 //removes repeating objects in arrays compared to the potential match
-                [matchDataGenres removeObjectsInArray:[potentialMatch valueForKey:@"genres"]];
-                [matchDataTracks removeObjectsInArray:[potentialMatch valueForKey:@"tracks"]];
-                [matchDataAlbums removeObjectsInArray:[potentialMatch valueForKey:@"albums"]];
-                [matchDataArtists removeObjectsInArray:[potentialMatch valueForKey:@"artists"]];
+                [matchDataGenresCopy removeObjectsInArray:[potentialMatch valueForKey:@"genres"]];
+                [matchDataTracksCopy removeObjectsInArray:[potentialMatch valueForKey:@"tracks"]];
+                [matchDataAlbumsCopy removeObjectsInArray:[potentialMatch valueForKey:@"albums"]];
+                [matchDataArtistsCopy removeObjectsInArray:[potentialMatch valueForKey:@"artists"]];
                 //gets the new count of the array
-                double newGenreCount = [matchDataGenres count];
-                double newAlbumCount = [matchDataAlbums count];
-                double newTrackCount = [matchDataTracks count];
-                double newArtistCount = [matchDataArtists count];
+                double newGenreCount = [matchDataGenresCopy count];
+                double newAlbumCount = [matchDataAlbumsCopy count];
+                double newTrackCount = [matchDataTracksCopy count];
+                double newArtistCount = [matchDataArtistsCopy count];
                 //generates a integer to determine how similar the two user's music taste is - 0 for similar and 1 for not similar at all
-                spotifyMatch = (((newGenreCount/genreCount)*0.2) + ((newTrackCount/trackCount)*0.3) + ((newAlbumCount/albumCount)*0.2) + ((newArtistCount/artistCount)*0.3));
+                if (genreCount == 0 || albumCount == 0 || trackCount == 0 || albumCount == 0){
+                    //To avoid dividing by zero if any of the counts are zero we make the Spotify match automatically 1
+                    self.spotifyMatch = 1;
+                }else{
+                self.spotifyMatch = (((newGenreCount/genreCount)*0.2) + ((newTrackCount/trackCount)*0.3) + ((newAlbumCount/albumCount)*0.2) + ((newArtistCount/artistCount)*0.3));
+                }
```

Comment:
It's a good practice to regularly format the code in the file using XCode as we discussed. (Notice the indentation of line 77.)